### DISCUSSION TITLE
How to avoid the BR-BT-01501-0107 error in a contract modification with more than one change?

### DISCUSSION THREAD

**Discussion by sorliden** [OP-TED/eForms-SDK#419] [Q&A] [answered]
I have a contract modification with two changes in different sections. But I still get the BR-BT-01501-0107 "_For any Contract Modification, the contract references (i.e. BT-1501(s)-Contract like 'CON-xxxx') is unique in the notice (i.e. there is at most one contract modification referring to a given contract)_" error.
Both section references ("PROCEDURE" and "RES-0001") are unique. 
Is the assert "_{ND-ContractModification} ${BT-1501(s)-Contract[BT-1501(s)-Contract like '^CON-\\d{4}$'] is unique in /BT-1501(s)-Contract}_" wrong?
I don't even have one 'CON-xxxx' section.
`          <efac:ContractModification>
            <efbc:ChangedNoticeIdentifier>d45e9175-a3bb-4e8e-bbc7-ea8103fa865c-01</efbc:ChangedNoticeIdentifier>
            <efac:Change>
              <efbc:ChangedSectionIdentifier>PROCEDURE</efbc:ChangedSectionIdentifier>
              <efbc:ChangeDescription languageID="ENG">Description of title modification.</efbc:ChangeDescription>
            </efac:Change>
            <efac:Change>
              <efbc:ChangedSectionIdentifier>RES-0001</efbc:ChangedSectionIdentifier>
              <efbc:ChangeDescription languageID="ENG">Description of modification.</efbc:ChangeDescription>
            </efac:Change>
            <efac:ChangeReason>
              <cbc:ReasonCode listName="modification-justification">mod-rev</cbc:ReasonCode>
              <efbc:ReasonDescription languageID="ENG">Reason for modification descriptionx</efbc:ReasonDescription>
            </efac:ChangeReason>
          </efac:ContractModification>`

**Comment by sorliden** [comment]
Here is the validation problem reproduced in eNotices2.
Both BR-BT-01501-0106 "_For a given Contract Modification, amongst all the internal references (BT-1501(s)-Contract), there is at least one to an existing contract_" (at least one "CON-xxxx" change section is needed) and
BR-BT-01501-0107 "_For any Contract Modification, the contract references (i.e. BT-1501(s)-Contract like "CON-xxxx") is unique in the notice (i.e. there is at most one contract modification referring to a given contract)_" (not more than one of each "CON-xxxx" change section is allowed).
Which is a contradiction. BR-BT-01501-0106 is true but BR-BT-01501-0107 is not.
<img width="745" alt="image" src="https://user-images.githubusercontent.com/43994636/233561613-ecf6c34d-a932-409c-888f-10a21610043d.png">

**Reply by YvesJo** [reply, marked as answer]
Hi,

BR-BT-01501-0106:
Contract modifications are about contracts and there is a need to identify the contract impacted. This is done using the Contract technical identifier (i.e. CON-xxxx) and any GR-ContractModification-Modification has to have at least one of these identifiers.
When the modification applies to multiple contracts, they could be referred to in the same GR-ContractModification-Modification

BR-BT-01501-0107:
Modifications related to a given contract should not be dispersed in multiple GR-ContractModification-Modification, hence for a given (modified) contract there should be a unique reference to it.

**Reply by sorliden** [reply]
So, it shouldn't be possible to have both errors at the same time.

**Reply by YvesJo** [reply]
Both rules are evaluated separately and could both be triggered for a given notice when no reference to a contract is specified in a contractmodification-modification group and a same contract is referenced in at least two other different contractmodification-modification groups. (scenario which would require at least 3 of these groups).

**Comment by sorliden** [comment]
It would be great if someone responsible for the validation rules could please do a reflection about this problem since it seems to be a stopper for the contract modification notices.
